Sources/EmbraceCore/Options/Embrace+Options.swift

```diff
@@ -54,22 +54,50 @@ extension Embrace {
             self.logLevel = logLevel
             self.export = export
         }
+
+        /// Initializer for `Embrace.Options` that does not require an appId.
+        /// Use this initializer if you don't want the SDK to send data to Embrace's servers.
+        /// You must provide your own `OpenTelemetryExport` on this mode.
+        ///
+        /// - Parameters:
+        ///   - platform: `Platform` in which the app will run. Defaults to `.iOS`.
+        ///   - captureServices: The `CaptureServices` to be installed.
+        ///   - crashReporter: The `CrashReporter` to be installed.
+        ///   - logLevel: The `LogLevel` to use for console logs.
+        ///   - export: `OpenTelemetryExport` object to export telemetry outside of the Embrace backend.
+        @objc public init(
+            platform: Platform = .default,
+            captureServices: [CaptureService],
+            crashReporter: CrashReporter?,
+            logLevel: LogLevel = .default,
+            export: OpenTelemetryExport
+        ) {
+            self.appId = "no_id"
```

Comment:
We need to audit the side effects of this change. I think it may be better to declare `appId` as Optional such that we have a clear understanding of the state of the SDK. 

There are some areas in the SDK that we should rename to update our assumptions. For instance, the `EmbraceFilePathProvider` takes an app_group and app_id to build the file path where data is stored. We should update this parameter to mark it as an opaque storage "partition"